dashboard/app/controllers/api/v1/sections_controller.rb

```diff
@@ -103,6 +104,12 @@ def student_script_ids
     render json: {studentScriptIds: @section.student_script_ids}
   end
 
+  # GET /api/v1/sections/membership
+  # Get the set of sections that the current user is enrolled in.
+  def membership
+    render json: {sections: @sections}
```

Comment:
I think I am missing a step here. How is `@sections` being converted into object that can be rendered as JSON? Are we somehow implicitly calling `section.summarize`?

Reply:
Thanks Maddie. How does that array of sections (which are `ActiveModel`s) then get serialized into something that can be rendered as json? is there an `ActiveModel::Serializer` hiding somewhere?

Reply:
oh, i see, i think i misunderstood your question. i'm not sure where `sections` are being serialized, to be honest. if i look at the response from this endpoint, it is serialized correctly:

```
[
  {
    "id"=>5700, 
    "user_id"=>4026, 
    "name"=>"Section 126", 
    "created_at"=>"2018-05-17T00:50:58.000Z", 
    "updated_at"=>"2018-05-17T00:50:58.000Z", 
    "code"=>"PWCDPL", 
    "script_id"=>6, 
    "course_id"=>nil, 
    "grade"=>nil, 
    "deleted_at"=>nil, 
    "stage_extras"=>false, 
    "section_type"=>nil, 
    "first_activity_at"=>nil, 
    "pairing_allowed"=>true, 
    "sharing_disabled"=>false, 
    "hidden"=>false
  }
]
```

Reply:
It looks like the existing `index` action calls [Section.summarize](https://github.com/code-dot-org/code-dot-org/blob/80d34bf7a8d081de9230b4355092ac509aae56aa/dashboard/app/models/sections/section.rb#L223-L222) for serialization, so I'd guess we probably want to do the same thing here. Default model serialization (as above) may not have the formatting the client expects, and may contain fields we don't need, such as `created_at` and `updated_at`. 

Along with writing new API controllers in `/api/v1`, we are moving toward using active model serializers for the associated JSON, so yes for best practice I'd add a serializer, but since the `summarize` method is already in use here, I think it's fine to stick with that for now.

Reply:
that sounds good, although the consumer of this endpoint (there's only one) only uses `name` and `id` on `section`, so would it be worthwhile to write a custom serializer (similar to `summarize` rather than creating an `ActiveModel::Serializer`) to lighten the payload?

Reply:
Ah yes, in that case since we're selecting specific fields, I'd go with a serializer specific to this usage with an appropriate name.

Reply:
That is, I'd be ok with either one - a method or an ActiveModel::Serializer, since we already have a method here, with a slight preference to ActiveModel::Serializer. 

Worth noting - our Serializer gem is very outdated and has some known perf issues. It's on the tech debt list, high on my personal list. Once we upgrade it and work out the kinks, I'll be much more in favor of putting all these JSON presenters there.

Reply:
I'm pretty sure the default serialization behavior for json render on an ActiveRecord model is to call `to_json` on the model, which AFAIK dumps everything in the `attributes` hash (essentially the DB columns, but it can have other things too) to json.

Reply:
Playing around, it seems `model.to_json` is slightly different than `model.attributes.to_json`. In the case of `Section` it hides `login_type` which is a db column, and an attribute, but here designated as the [inheritance column](https://github.com/code-dot-org/code-dot-org/blob/80d34bf7a8d081de9230b4355092ac509aae56aa/dashboard/app/models/sections/section.rb#L35-L35) for single-table-inheritance. 

In any case, `model.to_json` is the default serialization, and it looks like ActiveRecord hides some reserved attributes, but serializes the rest.

Reply:
Excellent, thanks for explaining Andrew!

Reply:
thank you both! this is really helpful. it sounds like the best option right now is to add an `ActiveModel::Serializer` for `Section`, so i'll go ahead and add that now